closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -314,6 +319,22 @@ int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
         return 0;
     }
 
+    /*
+     * This method is for CipherCore to insert the remainder of its buffer
+     * into the ibuffer before a encryptFinal(ByteBuffer, ByteBuffer) operation
+     */
+    int encrypt(byte[] in, int inOfs, int len) {
+        if (len > 0) {
+            // store internally until encryptFinal
+            ArrayUtil.nullAndBoundsCheck(in, inOfs, len);
```

Comment:
I think this check should precede the `if` statement.

Reply:
Isn't it faster to compare integers (`if (len > 0)`) than doing `ArrayUtil.nullAndBoundsCheck`? I'm just thinking that the common case would be that the user passes zero as the third argument (`len`) when passing `null` for the first argument (`in`). So it makes sense to first check the length.

Reply:
If the length is negative, you've suppressed an exception that should have been thrown.

Reply:
Okay, I changed it.